buster/docparser.py

```diff
@@ -100,12 +100,30 @@ def get_all_subsections(soup: BeautifulSoup) -> tuple[list[str], list[str], list
     return documents_df
 
 
+def get_file_extension(filepath: str) -> str:
+    return os.path.splitext(filepath)[1]
+
+
 def write_documents(filepath: str, documents_df: pd.DataFrame):
-    documents_df.to_csv(filepath, index=False)
+    ext = get_file_extension(filepath)
+
+    if ext == ".csv":
+        documents_df.to_csv(filepath, index=False)
+    elif ext in [".gz", ".bz2", ".zip", ".xz", ".zst", ".tar", ".tar.gz", ".tar.xz", ".tar.bz2"]:
```

Comment:
can we put this list in a toplevel var?

`PICKLE_EXTENSIONS = [".gz", ".bz2", ".zip", ".xz", ".zst", ".tar", ".tar.gz", ".tar.xz", ".tar.bz2"]`